src/pages/SignUp/__test__/Signup.test.tsx

```diff
@@ -0,0 +1,424 @@
+    render(
+      <MemoryRouter>
+        <SignUp></SignUp>
+      </MemoryRouter>
+    );
+
+    const inputAddress = screen.getByRole("textbox", {
+      name: /address/i,
+    });
+    act(() => {
+      userEvent.type(inputAddress, "India");
+    });
+
+    expect(
+      screen.getByRole("textbox", {
+        name: /address/i,
+      })
+    ).toHaveValue("India");
+  });
+
+  it("checks for valid value in pincode input field", () => {
+    render(
+      <MemoryRouter>
+        <SignUp></SignUp>
+      </MemoryRouter>
+    );
+
+    const inputPincode = screen.getByRole("spinbutton", {
+      name: /pincode/i,
+    });
+    act(() => {
+      userEvent.type(inputPincode, "123456");
+    });
+
+    expect(
+      screen.getByRole("spinbutton", {
+        name: /pincode/i,
+      })
+    ).toHaveValue(123456);
+  });
+
+  it("checks for valid value in email input field", () => {
+    render(
+      <MemoryRouter>
+        <SignUp></SignUp>
+      </MemoryRouter>
+    );
+
+    const inputEmail = screen.getByRole("textbox", {
+      name: /email/i,
+    });
+
+    act(() => {
+      userEvent.type(inputEmail, "manas@gmail.com");
+    });
+
+    expect(
+      screen.getByRole("textbox", {
+        name: /email/i,
+      })
+    ).toHaveValue("manas@gmail.com");
+  });
+
+  it("checks for valid value in password input field", () => {
+    render(
+      <MemoryRouter>
+        <SignUp></SignUp>
+      </MemoryRouter>
+    );
+
+    const inputPassword = screen.getByPlaceholderText(/enter password\.\.\./i);
+
+    act(() => {
+      userEvent.type(inputPassword, "Aasdasd12/@1");
+    });
+
+    expect(screen.getByPlaceholderText(/enter password\.\.\./i)).toHaveValue(
+      "Aasdasd12/@1"
+    );
+  });
+
+  it("checks for valid value in confirm password input field", () => {
+    render(
+      <MemoryRouter>
+        <SignUp></SignUp>
+      </MemoryRouter>
+    );
+
+    const inputConfirmPassword = screen.getByPlaceholderText(
+      /re\-enter the password/i
+    );
+
+    act(() => {
+      userEvent.type(inputConfirmPassword, "Aasdasd12/@1");
+    });
+
+    expect(screen.getByPlaceholderText(/re\-enter the password/i)).toHaveValue(
+      "Aasdasd12/@1"
+    );
+  });
+
+  it("checks for error message when name is not added", async () => {
+    render(
+      <MemoryRouter>
+        <SignUp></SignUp>
+      </MemoryRouter>
+    );
+    fireEvent.input(screen.getByRole("textbox", { name: /mobile number/i }), {
+      target: {
+        value: 9876543212,
+      },
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /email/i }), {
+      target: {
+        value: "Manas@Manas.com",
+      },
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /address/i }), {
+      target: {
+        value: "address",
+      },
+    });
+
+    fireEvent.submit(
+      screen.getByRole("button", {
+        name: /submit/i,
+      })
+    );
+
+    expect(await screen.findAllByTestId("name-error")).toHaveLength(1);
```

Comment:
suggestion: check for the exact validation message instead of toHaveLength using findByText or queryByText.